src/cluster_argument_parsing.rs

```diff
@@ -948,6 +926,87 @@ pub fn generate_galah_clusterer<'a>(
                     },
                     _ => panic!("Programming error"),
                 },
+                clusterer: match clap_matches
+                    .get_one::<String>(&argument_definition.dereplication_cluster_method_argument)
+                    .unwrap()
+                    .as_str()
+                {
+                    "fastani" => Clusterer::Fastani {
+                        threshold: parse_percentage(
+                            clap_matches,
+                            &argument_definition.dereplication_ani_argument,
+                        )
+                        .unwrap_or_else(|_| {
+                            panic!(
+                                "Failed to parse ani {:?}",
+                                clap_matches.get_one::<f32>(
+                                    &argument_definition.dereplication_ani_argument
+                                )
+                            )
+                        })
+                        .unwrap_or_else(|| {
```

Comment:
Is there a reason that this happens twice?

Reply:
Because parse_percentage returns `Result<Option<f32>`. I think that's OK, if a little confusing.